aws-blog-event-driven-batch-analytics/src/main/java/com/amazonaws/bigdatablog/edba/LambdaContainer.java

```diff
@@ -0,0 +1,249 @@
+			String stepState = emr.describeStep(stepReq).getStep().getStatus().getState();
+			
+				if(stepState.equals(StepState.COMPLETED.toString())){
+					ps.setString(1,StepState.COMPLETED.toString());
+				}else if (stepState.equals(StepState.FAILED.toString())){
+					ps.setString(1,StepState.FAILED.toString());					
+				}
+				ps.setString(2,openStepsRS.getString("job_config_id"));
+				ps.addBatch();
+		}
+		
+		ps.executeBatch();
+		ps.close();
+		conn.close();
+	}
+	
+	// adds EMR step
+	protected String fireEMRJob(String paramsStr,String clusterId){
+		StepFactory stepFactory = new StepFactory();
+		AmazonElasticMapReduceClient emr = new AmazonElasticMapReduceClient();
+		emr.setRegion(Region.getRegion(Regions.US_EAST_1));
+		Application sparkConfig = new Application()
+				.withName("Spark");
+		
+		String[] params = paramsStr.split(",");
+		StepConfig enabledebugging = new StepConfig()
+				.withName("Enable debugging")
+				.withActionOnFailure("TERMINATE_JOB_FLOW")
+				.withHadoopJarStep(stepFactory.newEnableDebuggingStep());
+		
+		HadoopJarStepConfig sparkStepConf = new HadoopJarStepConfig()
+				.withJar("command-runner.jar")
+				.withArgs(params);	
+		
+		final StepConfig sparkStep = new StepConfig()
+				.withName("Spark Step")
+				.withActionOnFailure("CONTINUE")
+				.withHadoopJarStep(sparkStepConf);
+
+		
+		AddJobFlowStepsRequest request = new AddJobFlowStepsRequest(clusterId)
+				.withSteps(new ArrayList<StepConfig>(){{add(sparkStep);}});
+				
+
+		AddJobFlowStepsResult result = emr.addJobFlowSteps(request);
+		return result.getStepIds().get(0);
+	}
+	
+	protected List<String> getActiveClusterId() throws Exception{
+		AmazonElasticMapReduceClient emrClient = new AmazonElasticMapReduceClient();
+		List<String> waitingClusters = new ArrayList<String>();
+		ListClustersResult clusterResult = emrClient.listClusters(new ListClustersRequest().withClusterStates(ClusterState.WAITING));
+		for( ClusterSummary cluster : clusterResult.getClusters()){
+				waitingClusters.add(cluster.getId());
+		}
+		return waitingClusters;
+		
+	}
+	/**
+	 * Checks whether additional criteria returned a non empty resultset.
+	 */
+	protected boolean isAdditionalCriteriaPassed(String sql, Connection conn) throws Exception{
+		if(StringUtils.isNullOrEmpty(sql)){
+			return true;
+		}
+		ResultSet rs = conn.createStatement().executeQuery(sql);
+		if(!rs.next()){
+			return false; // Empty Resultset
+		}
+		return true;
+	}
+	
+
+	static{
+		try{
+			Class.forName("com.mysql.jdbc.Driver").newInstance();
+		}catch(Exception ce){
+			ce.printStackTrace();
+		}
+	}
+	
+		final  String conditionFetchSQL = "select job_config_id, job_params,job_input_pattern,coalesce(last_run_timestamp,now() - interval 10 year) as last_run_timestamp,job_addl_criteria,coalesce(job_min_file_count,0) as job_min_file_count from edbaconfig.aggrjobconfiguration where last_exec_status is null or last_exec_status <> 'RUNNING'";
```

Comment:
I'd really consider putting these queries into a properties file, for simplicity of changes over time.